### PR TITLE
Updates to how we subscribe to websocket changes

### PR DESCRIPTION
**Pull request by atavism**

### CODE REVIEW

**Comment by atavism** [comment]
@jigar-f This one is still a work-in-progress, but feel free to take it for a spin and let me know how it works for you  if you get a chance.

**Comment by jigar-f** [comment]
@atavism Thanks, I'll take it for a spin next week.

**Comment by atavism** [comment]
@jigar-f Thanks!! This one is working really well for me (there's a couple areas I haven't tested yet like payment redirect and reporting an issue) so I'm curious how it works for you

File: desktop/lib.go [hunk 1]
```diff
@@ -76,6 +74,24 @@ var issueMap = map[string]string{
 	"Other":                       "9",
 }
 
+/*func initLantern() *app.App {
+		flags         := flashlight.ParseFlags()
```
Comment (jigar-f) [inline desktop/lib.go:78]: @atavism Can we remove this commented code?

File: lib/common/session_model.dart [hunk 2]
```diff
@@ -63,16 +71,20 @@ class SessionModel extends Model {
...
     } else {
-      country = ffiValueNotifier(ffiLang, 'lang', 'US');
-      proxyAvailable = ffiValueNotifier(
-        ffiSucceedingProxy,
-        'hasSucceedingProxy',
-        false,
-      );
-      userEmail = ffiValueNotifier(ffiEmailAddress, 'emailAddress', "");
-      proUserNotifier = ffiValueNotifier(ffiProUser, 'prouser', false);
+      configNotifier = ValueNotifier<ConfigOptions?>(null);
```
Comment (jigar-f) [inline lib/common/session_model.dart:74]: So now, we are updating notifier values from the WebSocket class

File: lib/common/websocket_subscriber.dart [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import 'common.dart';
+import 'common_desktop.dart';
+import 'package:collection/collection.dart';
+import 'package:lantern/plans/utils.dart';
+import 'package:fixnum/fixnum.dart';
+
+typedef void WebsocketChange();
+
+enum WebsocketMessage {
```
Comment (jigar-f) [inline lib/common/websocket_subscriber.dart:9]: Can change `WebsocketMessage` to `_WebsocketMessageType` also let's add _, I don't think we want to export the enum to anywhere except this class.

**Review by jigar-f — APPROVED** [review]
Changes look good, Just added a few comments.

File: lib/common/ffi_subscriber.dart [hunk 4]
```diff
@@ -3,12 +3,6 @@ import 'package:web_socket_channel/status.dart' as status;
...
 import 'common_desktop.dart';
 
-extension BoolParsing on String {
-  bool parseBool() {
-    return this.toLowerCase() == 'true';
-  }
-}
-
 class FfiValueNotifier<T> extends SubscribedNotifier<T?> {
   FfiValueNotifier(
```
Comment (jigar-f) [inline lib/common/ffi_subscriber.dart:7]: I think we can remove most of the arguments now, Since we are using only `defaultValue` and `removeFromCache`

**Comment by jigar-f** [comment]
I see there are a few places in sessionModel we are using `ffiValueBuilder`, I think we need migrated that as well

File: lib/common/websocket_subscriber.dart (hunk 3 above)
Comment (atavism) [reply lib/common/websocket_subscriber.dart:16]: Done

File: desktop/lib.go (hunk 1 above)
Comment (atavism) [reply desktop/lib.go]: Done

File: lib/common/ffi_subscriber.dart (hunk 4 above)
Comment (atavism) [reply lib/common/ffi_subscriber.dart:7]: Done

**Comment by atavism** [comment]
> I see there are a few places in sessionModel we are using ffiValueBuilder, I think we need migrated that as well

I removed the remaining instances of it except for chat where we are just setting some default values

**Pull request merged**